lib/wallet/src/Cardano/Wallet/DB/Store/Meta/Layer.hs

```diff
@@ -0,0 +1,132 @@
+import Cardano.Wallet.DB.Sqlite.Types
+    ( TxId (..) )
+import Cardano.Wallet.DB.Store.Meta.Model
+    ( DeltaTxMetaHistory, TxMetaHistory (..) )
+import Cardano.Wallet.DB.Store.Meta.Store
+    ( mkStoreMetaTransactions )
+import Cardano.Wallet.DB.Store.QueryStore
+    ( Query (..), QueryStore (..) )
+import Cardano.Wallet.Primitive.Types
+    ( Range (..), SortOrder (..) )
+import Data.Foldable
+    ( find, toList )
+import Data.List
+    ( sortOn )
+import Data.Maybe
+    ( catMaybes )
+import Data.Ord
+    ( Down (..) )
+import Data.Set
+    ( Set )
+import Database.Persist.Sql
+    ( Entity (entityVal)
+    , Filter
+    , PersistQueryRead (selectFirst)
+    , SelectOpt (..)
+    , SqlPersistT
+    , selectList
+    , (<=.)
+    , (==.)
+    , (>=.)
+    )
+import GHC.Natural
+    ( Natural )
+
+import qualified Cardano.Wallet.DB.Sqlite.Schema as DB
+import qualified Cardano.Wallet.Primitive.Types as W
+import qualified Data.Set as Set
+
+
+{-----------------------------------------------------------------------------
+    DB for 'TxMeta'
+------------------------------------------------------------------------------}
+
+-- | A query to the 'TxMeta' database.
+data QueryTxMeta b where
+    GetAll
+        :: Range SlotNo
+        -> Maybe Natural
+        -> SortOrder
+        -> QueryTxMeta [DB.TxMeta]
+    GetOne :: TxId -> QueryTxMeta (Maybe DB.TxMeta)
+
+filterMetas :: Range SlotNo -> [Filter TxMeta]
+filterMetas (Range Nothing Nothing) = []
+filterMetas (Range (Just low) Nothing) = [TxMetaSlot >=. low]
+filterMetas (Range  Nothing (Just high)) = [TxMetaSlot <=. high]
+filterMetas (Range (Just low) (Just high))
+    = [TxMetaSlot >=. low, TxMetaSlot <=. high]
+
+limitMetas :: Maybe Natural -> [SelectOpt record]
+limitMetas Nothing = []
+limitMetas (Just l) = [LimitTo $ fromIntegral l]
+
+orderMetas :: SortOrder -> [SelectOpt TxMeta]
+orderMetas Ascending = [Asc TxMetaSlot, Asc TxMetaTxId]
+orderMetas Descending = [Desc TxMetaSlot, Asc TxMetaTxId]
+
+-- | A 'QueryStore' for 'TxMeta'.
+mkQueryStoreTxMeta :: QueryStore (SqlPersistT IO) QueryTxMeta DeltaTxMetaHistory
+mkQueryStoreTxMeta =
+    QueryStore
+        { queryS = \case
+            GetAll range limit order ->
+                fmap entityVal <$>
+                    selectList @DB.TxMeta
+                        (filterMetas range)
+                        (limitMetas limit <> orderMetas order)
+            GetOne txId ->
+                fmap entityVal <$>
+                    selectFirst @_ @_ @DB.TxMeta
+                        [TxMetaTxId ==. txId]
+                        []
+        , store = mkStoreMetaTransactions
+        }
+
+-- | Get all 'TxId's after a given 'SlotNo'.
+getMetasAfterSlot
+    :: QueryStore (SqlPersistT IO) QueryTxMeta DeltaTxMetaHistory
+    -> SlotNo
+    -> SqlPersistT IO (Set TxId)
+getMetasAfterSlot q s =
+    fmap (foldMap (Set.singleton . txMetaTxId))
+        $ queryS q
+        $ GetAll
+            (Range (Just s) Nothing)
```

Comment:
Wait, there seems to be an off-by-one-error here. 😳 

The name `getMetasAfterSlot` suggests that precisely the `meta`s whith `slot meta > s` are returned, but the condition for `GetAll` is actually `sloto meta >= s`:

```hs
filterMetas (Range (Just low) Nothing) = [TxMetaSlot >=. low]
```

Moreover, this affects the `Store` — see comment there.